src/Shape/Shape.js

```diff
@@ -407,14 +411,17 @@ export default class Shape extends BaseClass {
             for (let l = 0; l < labels.length; l++) {
 
               const b = bounds.constructor === Array ? bounds[l] : Object.assign({}, bounds);
+              const r = d.labelConfig && d.labelConfig.rotate ? d.labelConfig.rotate + this._rotate(d, i) : bounds.angle !== undefined ? bounds.angle + this._rotate(d, i) : this._rotate(d, i);
```

Comment:
This line is crazy long. Since it's essentially just adding the `this._rotate` value to another value (which falls back to zero), can you separate out the logic for getting that first value into a line above this, and then just add the rotation to that?